draft-specifications/CapTP Specification.md

```diff
@@ -492,13 +492,16 @@ parties MUST send upon a new connection. The operation looks like this:
                   acceptable-location-sig>  ; CapTP signature
 ```
 
-It's important that we only have one bidirectional connection between a CapTP
-session. Before trying to connect to a peer, it's important to check that a
-connection is not already open before proceeding. There is a race condition
-called the "crossed hellos" problem whereby each side tries to open a new
-connection at the same time and only one of these connection attempts must
-succeed. The way to check which attempt should succeed is to sort the serialized
-`session-pubkey`, the highest of the two wins.
+An important aspect of CapTP is that only one active session between two peers
+should exist. This allows peers to equality checks against objects from a remote
+peer and prevents [3rd Party Handoffs](#third-party-handoffs) from where the
+receiver and exporter are on the same peer.
+
+There are several mechanisms put in place to ensure that only one session exists
+between two peers. The first is that a CapTP implementation MUST check if it has
+an active session with a given peer, before trying to establish a new
+connection. If the peer already has an active session, that session MUST be used
+instead of creating a new one.
```

Comment:
Out of curiosity, does the spec ever explain what "session with a given peer" means exactly? For example, if you have two ways of reaching a peer over two different netlayers, we consider those two different peers, right? I feel like this is a question that someone reading/implementing OCapN would have.

Reply:
It doesn't. Since we don't currently support using multiple netlayers together, the union of the `transport` and `designator` is the deciding factor. This is an underspecified part of the spec and I suspect we do want to support multiple netlayers which probably requires some global (to all netlayers) set of keys and identity.